src/util/util.js

```diff
@@ -87,6 +103,28 @@ export function getBounds(points: Point[]): { min: Point, max: Point} {
     };
 }
 
+/**
+ * Returns the 2D square distance between an AABB defined by min and max and a point.
+ * If point is null or undefined, the AABB distance from the origin (0,0) is returned.
+ *
+ * @param {Point} min The minimum extent of the AABB.
+ * @param {Point} max The maximum extent of the AABB.
+ * @param {Point} point The point to compute the distance from, may be undefined.
+ * @returns {number} The square distance from the AABB, 0.0 if the AABB contains the point.
+ */
+export function getAABBPointSquareDist(min: Point, max: Point, point: ?Point): number {
```

Comment:
I'm not signed into shadertoy so it's hard to share, but here's an alternate, maybe simpler implementation that returns the squared distance to the box (including sides) rather than the minimum of the squared distance to one of the four corners:

```glsl
// Return the square of the SDF of an aabb:
float sdAabbSq( in vec2 p, in vec2 b1, in vec2 b2 )
{
    vec2 c = 0.5 * (b2 + b1);  // center
    vec2 s = 0.5 * (b2 - b1);  // size
    vec2 d = max(abs(p - c) - s, 0.0);
    return dot(d, d);
}
```

<img width="1220" alt="Screen Shot 2021-05-06 at 10 47 37 AM" src="https://user-images.githubusercontent.com/572717/117342829-7fd5b880-ae58-11eb-9c64-ad9620d59dad.png">

Otherwise maybe we could call this function `getAABBPointCornerSquaredDistance` to avoid potential confusion in what it computes.

Reply:
Okay, can't propose a change like this without just conveying the code. The main thing that jumped out was that it subtly computes the distance to one of the four corners and not to the sides, in a way which could give subtly wrong answers, depending on whether you expect an upper or lower bound. Here's a JS implementation of the above along with a visual check:

<img width="444" alt="Screen Shot 2021-05-06 at 11 12 50 AM" src="https://user-images.githubusercontent.com/572717/117345991-093aba00-ae5c-11eb-9233-019533ce76f2.png">

https://observablehq.com/d/27f1b8671e22ea66

```js
function sqAabbDist(min, max, point) {
  let d = 0;
  for (let i = 0; i < 2; i++) {
    assert(min[i] < max[i], 'Invalid aabb min and max inputs, min[i] must be < max[i].');
    const p = point ? point[i] : 0;
    const c = 0.5 * (max[i] + min[i]);
    const s = 0.5 * (max[i] - min[i]);
    d += Math.pow(Math.max(Math.abs(p - c) - s, 0), 2);
  }
  return d;
}
```

Reply:
Oh no, facepalm. After all that, I realize I subtly misread the code and that is indeed what the original function computes. Apologies. Please disregard. 🤦